internal/provisioner/kops_provisioner_cluster.go

```diff
@@ -260,6 +273,24 @@ func (provisioner *KopsProvisioner) ProvisionCluster(cluster *model.Cluster, aws
 		return errors.Wrap(err, "failed to delete PodDisruptionBudget calico-typha")
 	}
 
+	// Need to remove two items from the metrics-server because the fields after the creation are imutable so the
+	// create/update does not work. We might want to refactor this in the future to avoid this
+	logger.Info("Cleaning up some metrics-server resources to reapply")
+	err = k8sClient.Clientset.CoreV1().Services("kube-system").Delete("metrics-server", &metav1.DeleteOptions{})
+	if k8sErrors.IsNotFound(err) {
+		logger.Info("Service metrics-server not found; skipping...")
+	} else if err != nil {
+		return errors.Wrap(err, "failed to delete service metrics-server")
+	}
+
+	logger.Info("Cleaning up some metrics-server resources to reapply")
+	err = k8sClient.KubeagClientSet.ApiregistrationV1beta1().APIServices().Delete("v1beta1.metrics.k8s.io", &metav1.DeleteOptions{})
+	if k8sErrors.IsNotFound(err) {
+		logger.Info("APIService v1beta1.metrics.k8s.io not found; skipping...")
+	} else if err != nil {
+		return errors.Wrap(err, "failed to delete APIService v1beta1.metrics.k8s.io")
+	}
```

Comment:
We really shouldn't need to check this on create as the cluster is being created for the first time right? Doesn't hurt to have it, but just thinking that this is probably safe to leave out.

Reply:
only if we update the manifests

Reply:
Okay, that makes sense!

Reply:
this function is called for create/upgrade and provision API calls